src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -897,41 +897,20 @@ internal void ProcessHandshakeSuccess()
                 size   -
                 output - Encrypted bytes
         --*/
-        internal SecurityStatusPal Encrypt(byte[] buffer, int offset, int size, ref byte[] output, out int resultSize)
+        internal SecurityStatusPal Encrypt(ReadOnlyMemory<byte> buffer, ref byte[] output, out int resultSize)
         {
             if (NetEventSource.IsEnabled)
             {
-                NetEventSource.Enter(this, buffer, offset, size);
-                NetEventSource.DumpBuffer(this, buffer, 0, Math.Min(buffer.Length, 128));
+                NetEventSource.Enter(this, buffer, 0, buffer.Length);
+                int eventLength = Math.Min(buffer.Length, 128);
+                NetEventSource.DumpBuffer(this, buffer.Slice(0, eventLength).ToArray(), 0, eventLength);
             }
 
             byte[] writeBuffer = output;
-
-            try
-            {
-                if (offset < 0 || offset > (buffer == null ? 0 : buffer.Length))
-                {
-                    throw new ArgumentOutOfRangeException(nameof(offset));
-                }
-
-                if (size < 0 || size > (buffer == null ? 0 : buffer.Length - offset))
-                {
-                    throw new ArgumentOutOfRangeException(nameof(size));
-                }
-
-                resultSize = 0;
-            }
-            catch (Exception e) when (!ExceptionCheck.IsFatal(e))
-            {
-                NetEventSource.Fail(this, "Arguments out of range.");
-                throw;
-            }
```

Comment:
Was all of this checking already done earlier in the call chain, prior to the array being stored into the memory?